include/seqan3/range/view/interleave.hpp

```diff
@@ -0,0 +1,352 @@
+// ============================================================================
+
+namespace seqan3::view
+{
+
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief A view that interleaves a given range into another range at regular intervals.
+ * \tparam urng_t The type of the range being processed.
+ * \tparam inserted_rng_t The type of the range being inserted.
+ * \param[in] urange The range being processed.
+ * \param[in] inserted_range The range being inserted.
+ * \param[in] step_size A value of size_type which indicates the interval to insert the inserted_range.
+ * \returns A range with the second range inserted at regular intervals. See below for properties of said range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view can be used to insert one range into another range at regular intervals.
+ *
+ * ### View properties
+ *
+ * | range concepts and reference_t  | `urng_t` (underlying range type)      | `rrng_t` (returned range type)                     |
+ * |---------------------------------|:-------------------------------------:|:--------------------------------------------------:|
+ * | std::ranges::InputRange         | *required*                            | *preserved*                                        |
+ * | std::ranges::ForwardRange       | *required*                            | *preserved*                                        |
+ * | std::ranges::BidirectionalRange |                                       | *preserved*                                        |
+ * | std::ranges::RandomAccessRange  |                                       | *preserved*                                        |
+ * | std::ranges::ContiguousRange    |                                       | *preserved*                                        |
+ * |                                 |                                       |                                                    |
+ * | std::ranges::ViewableRange      | *required*                            | *guaranteed*                                       |
+ * | std::ranges::View               |                                       | *guaranteed*                                       |
+ * | std::ranges::SizedRange         |                                       | *preserved*                                        |
+ * | std::ranges::CommonRange        |                                       | *preserved*                                        |
+ * | std::ranges::OutputRange        |                                       | *preserved*                                        |
+ * | seqan3::const_iterable_concept  |                                       | *preserved*                                        |
+ * |                                 |                                       |                                                    |
+ * | seqan3::reference_t             |                                       | seqan3::reference_t<urng_t>                                        |
```

Comment:
I would make two tables:

 | range concepts and reference_t  | `urng_t` (underlying range type)      | `rrng_t` (returned range type)                     |
  |---------------------------------|:-------------------------------------:|:--------------------------------------------------:|
  | std::ranges::InputRange         | *required*                            | *preserved*                                        |
  | std::ranges::ForwardRange       | *required*                            | *preserved*                                        |
  | std::ranges::BidirectionalRange |   *required*                  | *preserved*                                        |
  | std::ranges::RandomAccessRange  |     *required*                   | *preserved*                                        |
  | std::ranges::ContiguousRange    |                                       | *lost*                                          |
  |                                 |                                       |                                                    |
  | std::ranges::ViewableRange      | *required*                            | *guaranteed*                                       |
  | std::ranges::View               |                                       | *guaranteed*                                       |
  | std::ranges::SizedRange         |    *required*                  | *preserved*                                        |
  | std::ranges::CommonRange        |                                       | *preserved*                                        |
  | std::ranges::OutputRange        |                                       | *preserved*                                        |
  | seqan3::const_iterable_concept  |                                       | *preserved*                                        |
  |                                 |                                       |                                                    |
  | seqan3::reference_t             |                                       | seqan3::reference_t<urng_t>                                        |


If above requirements are not met, this adaptor forwards to `| ranges::view::split(step_size) | ranges::view::join(inserted_range)` which returns a view with the following properties:

 | range concepts and reference_t  | `urng_t` (underlying range type)      | `rrng_t` (returned range type)                     |
  |---------------------------------|:-------------------------------------:|:--------------------------------------------------:|
  | std::ranges::InputRange         | *required*                            | *preserved*                                        |
  | std::ranges::ForwardRange       | *required*                            | *lost*                                        |
  | std::ranges::BidirectionalRange |                         | *lost*                                           |
  | std::ranges::RandomAccessRange  |                          | *lost*                                          |
  | std::ranges::ContiguousRange    |                                       | *lost*                                           |
  |                                 |                                       |                                                    |
  | std::ranges::ViewableRange      | *required*                            | *guaranteed*                                       |
  | std::ranges::View               |                                       | *guaranteed*                                       |
  | std::ranges::SizedRange         |                         | *lost*                                           |
  | std::ranges::CommonRange        |                                       | *lost*                                          |
  | std::ranges::OutputRange        |                                       | *lost*                                          |
  | seqan3::const_iterable_concept  |                                       | *lost*                                          |
  |                                 |                                       |                                                    |
  | seqan3::reference_t             |                                       | seqan3::value_type_t<urng_t>                                        |

Reply:
Please make sure the spacing in the table is correct (likely not in my code snippet)

Reply:
this was not resolved